data_sender.py

```diff
@@ -21,24 +21,24 @@ def __init__(self):
     def convert_data(self):
         rot, trans = self.pose_calculator.get_values()
 
+        if trans[0] is None:
+            return 9999, 9999, 9999
+
         robot_x = trans[0]
-        robot_y = math.cos(self.CAMERA_TILT) * trans[1] - math.sin(self.CAMERA_TILT) * trans[2]
-        robot_z = math.sin(self.CAMERA_TILT) * trans[1] + math.sin(self.CAMERA_TILT) * trans[2]
+        robot_y = math.cos(math.radians(self.CAMERA_TILT)) * trans[1] - math.sin(math.radians(self.CAMERA_TILT)) * trans[2]
+        robot_z = math.sin(math.radians(self.CAMERA_TILT)) * trans[1] + math.sin(math.radians(self.CAMERA_TILT)) * trans[2]
 
-        target_y = math.cos(rot[2]) * trans[1] - math.sin(rot[2]) * (math.cos(rot[1]) * trans[2]) - math.sin(rot[1]) * trans[2]
-        target_x = math.cos(rot[1]) * trans[0] + math.sin(rot[1]) * trans[2]
-        target_z = math.sin(rot[2]) * trans[1] + math.cos(rot[2]) * (math.cos(rot[1]) * trans[2]) - math.sin(rot[1]) * trans[2]
+        target_x = trans[0] * math.cos(math.radians(rot[1])) - trans[0] * math.sin(math.radians(rot[1]))
+        target_y = trans[1] * math.cos(math.radians(rot[0])) + trans[1] * math.sin(math.radians(rot[0]))
+        target_z = trans[2] * (math.sin(math.radians(rot[1])) + math.sin(math.radians(rot[1]))) * (math.sin(math.radians(rot[0])) + math.cos(math.radians(rot[0])))
 
         angle = math.degrees(math.acos(target_y))
 
-        if trans[0] is None:
-            return 9999, 9999, 9999
-
         return int(100 * robot_x), int(100 * robot_z), 10 * round(angle, 1)
 
     def send_data(self):
         x, y, angle = self.convert_data()
-        self.s.write('S {:04d} {:04d} {:+05d} E\n'.format(x, y, angle))
+        self.s.write(bytes('S {:04d} {:04d} {:+05d} E'.format(x, y, angle), "utf-8"))
```

Comment:
What happened to the newline character?

Reply:
When we were testing yesterday, robot code said it would be easier to not have it. I can update the spec to include that.